projects/memapGui/src/main/java/fortiss/gui/PlotPanel.java

```diff
@@ -48,52 +61,50 @@ public PlotPanel() {
 	 */
-	public void addSeries(String seriesName, ArrayList<Double> series) {
+	public void addSeries(String seriesName, ArrayList<String> xData, ArrayList<Double> yData) {
 		if (!getChart().getSeriesMap().containsKey(seriesName)) {
-
-			/** this is commented out. It was required for the function below. 
-			 * Currently this funtion is not used.
-			Comparator<Double> comparator = new Comparator<Double>() {
-				@Override
-				public int compare(Double v1, Double v2) {
-					return v1.compareTo(v2);
-				}
-			};
+			
+			/** OLD code
+			//XYSeries seriesx = getChart().addSeries(seriesName, series);
+			//seriesx.setLabel(seriesName);
+			//seriesx.setMarker(SeriesMarkers.NONE);
+			//seriesx.setYAxisGroup(0);
 			*/
 			
-			//System.out.println("Plotten: ");
-			//System.out.println("SeriesName: " + seriesName);
+			CategorySeries chartSeries = getChart().addSeries(seriesName, xData, yData);					
+			chartSeries.setLabel(seriesName);
+			chartSeries.setMarker(SeriesMarkers.NONE);
 			
-			//String listString = series.stream().map(Object::toString)
-            //        .collect(Collectors.joining(", "));
+			min = 0.0;
+			max = 0.0;
 			
-			//System.out.println("Data: " + listString);
-
-			// Series with maximum value smaller than 0.5 in absolute value are plotted in a
-			// separate axis
+			System.out.println("************************************************");
 			
-			// FIXME the second yAxis on the right seems not to work therefore this is commented out
-			/*
-			if (Collections.max(series, comparator) < 0.5 && Collections.min(series, comparator) > -0.5
-					&& getChart().getSeriesMap().size() > 0) {
-				XYSeries seriesx = getChart().addSeries(seriesName, series);
-				seriesx.setLabel(seriesName + "(right)");
-				seriesx.setMarker(SeriesMarkers.NONE);
-				seriesx.setYAxisGroup(1);
-				getChart().getStyler().setYAxisMax(1, 0.5);
-				getChart().getStyler().setYAxisMin(1, -0.5);			
-			} else {
-			*/
+			getChart().getSeriesMap().forEach( (key, value) -> {
+				
+				//System.out.println(PlotPanel.class + " seriesName: " + key + " Yvalues: " + value.getYData());
+				//System.out.println(PlotPanel.class + " seriesName: " + key + " Xvalues: " + value.getXData());
+				//System.out.println(PlotPanel.class + " seriesName YAxsis: min=" + value.getYMin()+ "  max=" + value.getYMax());
+				
+				max  = (max > value.getYMax()) ? max : value.getYMax();
+				min  = (min < value.getYMin()) ? min : value.getYMin();
+			});
+			
+			if ( Math.abs(max-min) < 0.1) {
 				
-			XYSeries seriesx = getChart().addSeries(seriesName, series);
-			seriesx.setLabel(seriesName + "(left)");
-			seriesx.setMarker(SeriesMarkers.NONE);
-			seriesx.setYAxisGroup(0);
-			getChart().getStyler().setYAxisMax(0, null);
-			getChart().getStyler().setYAxisMin(0, null);
-			getChart().getStyler().setYAxisGroupPosition(0, YAxisPosition.Left);
-			//}
-		}		
+				System.out.println(PlotPanel.class + " same limits, changing them to larger size.");
```

Comment:
Is this console print intended for debugging? If so, we can remove it. If it is intended to give information to the user, using the log feature would be a better alternative.

Reply:
done